packages/engine/crawlers/sui/collectibles.ts

```diff
@@ -0,0 +1,151 @@
+import type {
+	JsonRpcProvider,
+	PaginatedObjectsResponse,
+	SuiObjectData,
+} from '@mysten/sui.js';
+import { Networks } from '@walless/core';
+import type { CollectibleDocument } from '@walless/store';
+
+export const getSuiCollectibles = async (
+	connection: JsonRpcProvider,
+	owner: string,
+) => {
+	const collectibleDocuments: CollectibleDocument[] = [];
+	let objectResponses: PaginatedObjectsResponse['data'] = [];
+	let nextCursor: PaginatedObjectsResponse['nextCursor'] | undefined;
+	let cursor: PaginatedObjectsResponse['nextCursor'] | undefined;
+	while (nextCursor !== null) {
+		const allOwnedObjects = await connection.getOwnedObjects({
+			owner,
+			cursor,
+			options: {
+				showType: true,
+				showOwner: true,
+				showDisplay: true,
+				showContent: true,
+			},
+		});
+
+		cursor = allOwnedObjects.nextCursor;
+
+		if (allOwnedObjects.hasNextPage && nextCursor !== cursor) {
+			nextCursor = cursor;
+		} else {
+			nextCursor = null;
+		}
+
+		objectResponses = [...objectResponses, ...allOwnedObjects.data];
+	}
+
+	objectResponses
+		.filter((objResponse) => {
+			if (objResponse.data) {
+				return isCollectible(objResponse.data);
+			}
+
+			return false;
+		})
```

Comment:
isCollectible should return false if arg is null/undefined. So it just `objectResponsed.filter(res => isCollectible(res.data))`